### PR TITLE
fix breaking distraction free

### PR DESCRIPTION
**Pull request by corentin-gautier**
## What?

This fixes an issue with the distraction free mode since changing the header overflow in #47177

## Why?

Using overflow hidden on .interface-interface-skeleton__header prevent the mouse from hovering the header and triggering the display when in distraction free mode

## How?

We move the styles to the .edit-post-header

## Testing Instructions

1. Open a Post or Page.
2. Enter distraction free mode
3. Move your mouse to the top of the screen
4. The header should appear

### CODE REVIEW

**Comment by corentin-gautier** [comment]
@ciampo @jasmussen Hi ! The changes that were made in #47177 prevent users from leaving the distraction free mode. (because of the overflow hidden on the header) 

Could you take a look at this fix ?

**Review by jasmussen — APPROVED** [review]
Thank you for a prompt fix! In quick tests, this is working for me, and it seems harmless.

Before you merge, can you please test also in the site editor? The nested block there might be called `edit-site-header`. 

GIF showing before, how the mouse cursor towards the top doesn't reveal the toolbar:

before

GIF of after, showing the toolbar appearing again:

after

**Comment by corentin-gautier** [comment]
@jasmussen Unless I'm mistaken the changes were only applied to the post editor. I took a look at the site editor but the DOM is widely different. I think we can merge this now and see later if we add the same behaviour on the site editor, what do you think ? Also I can't merge PRs ;)

**Comment by jasmussen** [comment]
If both editors work, we can revisit, yes.

**Comment by corentin-gautier** [comment]
@jasmussen Yes both work

**Pull request merged**
